### ISSUE TITLE
Azure blob output binding storage account level auth option

### ISSUE THREAD

**Issue by alicejgibbons** [dapr/components-contrib#3146]
## Describe the feature
Today, the Azure Blob storage output binding uses authentication mechanisms that sit at the blob container level.  This causes Dapr users to have to use a different blob storage component for every blob container that sits in the same Azure storage account which can mean managing many component manifests and many authentication keys in some cases.

The component implementation should add the option to authenticate at the Azure storage account level instead of at the container level. Since all calls to blob storage will still require a blob container name, it could be provided as metadata in the request to the Binding API instead of in the component manifest.

This would allow greater flexibility for Dapr users and allow for less management of component manifests and storage access keys.

## Release Note

RELEASE NOTE: **ADD:** Auth option for Azure Blob output binding at the storage account level.

**Comment by alicejgibbons** [comment]
not stale

**Comment by berndverst** [comment]
We will have to contemplate this carefully. The existing behavior must be retained as the default for security reason. What we do not want is to allow an application or person to read or write data from a container to which they should not have access. Request metadata is very insecure in this way.

*If* we go with enabling this use case, it would need to be exposed via a non-default, advanced option - e.g. a new metadata option called "useAccountLevelAuth" (bool). Only if this option is explicitly enabled would we then use the request metadata to infer the desired storage account.

@alicejgibbons can you provide user / customer evidence for this request? This adds complexity and is not without risk.

@fabistb would you like to speak toward this on behalf of your company?

**Comment by alicejgibbons** [comment]
@berndverst yes this has been requested by a Dapr end-user. I think your approach with an additional metadata options sounds good for backward compatibility.

**Comment by fabistb** [comment]
@berndverst  sorry for the late reply.

In my specific case an file / asset manage application has currently > 50 components.
This makes it more and more cumbersome to manage it. Enabling this functionality would make this easier.


The approach you described with the metadata as advanced option sounds good.
